product-catalog/test/productcatalog/services/ProductProjectionServiceTest.java

```diff
@@ -0,0 +1,55 @@
+package productcatalog.services;
+
+import common.categories.CategoryUtils;
+import common.products.ProductUtils;
+import io.sphere.sdk.categories.Category;
+import io.sphere.sdk.client.PlayJavaSphereClient;
+import io.sphere.sdk.client.SphereRequest;
+import io.sphere.sdk.json.JsonUtils;
+import io.sphere.sdk.products.ProductProjection;
+import io.sphere.sdk.queries.PagedQueryResult;
+import org.junit.Test;
+import play.libs.F;
+
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class ProductProjectionServiceTest {
+
+    private static final int DEFAULT_TIMEOUT = 1000;
+
+    @Test
+    public void testFindVariantBySku() throws Exception {
+        final ProductProjectionService service = new ProductProjectionService(null);
+        final ProductProjection product = JsonUtils.readObjectFromResource("product.json", ProductProjection.typeReference());
+        final String masterSku = "M0E20000000DSB9";
+        final String otherSku = "M0E20000000DSBA";
+
+        assertThat(service.findVariantBySku(product, masterSku).get().getSku()).contains(masterSku);
+        assertThat(service.findVariantBySku(product, otherSku).get().getSku()).contains(otherSku);
+        assertThat(service.findVariantBySku(product, "")).isEmpty();
+    }
+
+    @Test
+    public void testGetSuggestions() {
+        final PlayJavaSphereClient sphereClient = new PlayJavaSphereClient() {
```

Comment:
I'd move this implementation to a method, giving it a meaningful name about what kind of sphere client it is trying to implement. We should leave the tests clean and focused just on what it is important.